src/components/core/list/base/base-list.tsx

```diff
@@ -0,0 +1,20 @@
+import { forwardRef } from 'react';
+import { OrderedList, UnorderedList } from '../composed/derived-lists';
+import { ListItemProps } from '../types';
+
+const BaseList = forwardRef<HTMLUListElement | HTMLOListElement, ListItemProps>(({ type, children, ...rest }, ref) => {
+    if (type === 'numbered') {
```

Comment:
```suggestion
    if (type === ListType.NUMBERED) {
```
Check [Comment #1]. ☝🏼  This is how it would look like using enum based on the comment's recommendation. You can omit this code review if we only check for list type in this specific condition 😄

Reply:
I only use it in this specific condition and based on my answer to the first comment, can't change it to `enum`. I can actually change it to `enum` but it will brings some kinda overhead to the implementation since I have to convert the `enum` back to string literals.